testing/unittests/jlibtests.cpp

```diff
@@ -603,6 +603,74 @@ CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( JlibFileIOTestTiming, "JlibFileIOTestTimi
 CPPUNIT_TEST_SUITE_REGISTRATION( JlibFileIOTestStress );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( JlibFileIOTestTiming, "JlibFileIOTestStress" );
 
+/* =========================================================== */
+class JlibContainsRelPathsTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(JlibContainsRelPathsTest);
+        CPPUNIT_TEST(testWindowsPath);
+        CPPUNIT_TEST(testNonWindowsPath);
+    CPPUNIT_TEST_SUITE_END();
+
+    bool testContainsRelPaths(const char * path)
+    {
+        return containsRelPaths(path);
+    };
+public:
+    void testWindowsPath()
+    {
+        CPPUNIT_ASSERT(testContainsRelPaths("a\\b\\c") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths(".\\a\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("..\\a\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("...\\a\\b\\c") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\.\\a\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\..\\a\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\...\\a\\b\\c") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c\\.") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c\\..") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c\\...") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\.\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\..\\b\\c") == true);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\...\\b\\c") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c.d\\e") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c..d\e") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c...d\\e") == false);
+        CPPUNIT_ASSERT(testContainsRelPaths("\\a\\b\\c.d.e\\f") == false);
+
+        printf("Finished JlibContainsRelPathsTest for Windows Paths\n");
```

Comment:
don't think should have printf's here, you can run unittests with -v for progress.